src/components/ui/Button.tsx

```diff
@@ -0,0 +1,219 @@
+import {
+  alignItems,
+  backgroundClip,
+  backgroundColor,
+  backgroundImage,
+  borderColor,
+  borderRadius,
+  borderWidth,
+  boxShadow,
+  boxShadowColor,
+  brightness,
+  classnames,
+  cursor,
+  display,
+  flexDirection,
+  fontFamily,
+  fontSize,
+  fontWeight,
+  gradientColorStops,
+  justifyContent,
+  lineHeight,
+  opacity,
+  outlineStyle,
+  padding,
+  space,
+  textAlign,
+  textColor,
+  transitionDuration,
+  transitionProperty,
+  transitionTimingFunction,
+  width,
+} from '@/classnames/tailwind'
+import Arrow from '@/icons/Arrow'
+import Loading from '@/icons/Loading'
+import React from 'react'
+
+const commonClasses = (
+  type: ButtonType,
+  fullWidth?: boolean,
+  center?: boolean,
+  available?: boolean,
+  small?: boolean,
+  gradientFont?: boolean
+) =>
+  classnames(
+    display('flex'),
+    flexDirection('flex-row'),
+    justifyContent(center ? 'justify-center' : undefined),
+    alignItems('items-center'),
+    fontWeight('font-bold'),
+    fontFamily('font-primary'),
+    transitionProperty('transition-all'),
+    transitionTimingFunction('ease-in-out'),
+    transitionDuration('duration-100'),
+    cursor({ 'cursor-not-allowed': !available }),
+    outlineStyle('focus:outline-none'),
+    opacity({ 'opacity-50': !available }),
+    boxShadow({
+      'shadow-2xl': available,
+      'hover:shadow-lg': available,
+      'active:shadow-button-active': available && !gradientFont,
+    }),
+    width(fullWidth ? 'w-full' : 'w-fit'),
```

Comment:
`w-fit` usually works as default: 
![image](https://user-images.githubusercontent.com/49658988/189184495-dbd3df7d-3d34-4dbe-9b2c-bdcef7b87067.png)
So you can do `width({'w-full': fullWidth})`